src/Hyperion/PrintReport.hs

```diff
@@ -0,0 +1,24 @@
+{- This module is used for pretty printing the hyperion report.
+ - This is mostly put in a separate module to avoid conflicts in
+ - the imports of (<>) from Monoid an ansi-wl-pprint
+ -}
+
+module Hyperion.PrintReport (printReports) where
+
+import Control.Lens (view)
+import Control.Monad (forM_)
+import Data.Text (Text, unpack)
+import Data.HashMap.Strict (elems, HashMap)
+import Hyperion.Report
+import Text.PrettyPrint.ANSI.Leijen
+
+printReport :: Report -> IO ()
+printReport report = do
+    putDoc $ green (bold (text (unpack (view reportBenchName report)))) <> line
+    putDoc $
+       (indent 2 $ text ("Bench time: " ++ (show (view reportTimeInNanos report)) ++ "ns")) <> line
+
+printReports :: HashMap Text Report -> IO ()
+printReports report = do
+  _ <- forM_ (elems report) printReport
```

Comment:
Make `printReport` return a `Doc` and use [`mconcat`](http://hackage.haskell.org/package/base-4.9.1.0/docs/Data-Monoid.html#v:mconcat) here to assemble. So that you have a single `putDoc` for the entire output.